### DISCUSSION TITLE
Internal vs external documentation

### DISCUSSION THREAD

**Discussion by kryltsov** [markdoc/markdoc#448] [Q&A] [answered]
The question is mainly for Stripe and others who run internal and internal documentation with Markdoc.

Do you use a single repo for internal and external docs?

Do you make separate docs for the same system/API or use tags to generate internal or external docs?

In another thread, you mentioned that all documentation is in a single repo.  Do developers update docs and what are the cons and pros of having some parts of documentation to be stored within respective repos so it is easier for dev teams to review the docs changes?

If you keep them separate how do you sync them if you do? 

Thank you

**Comment by duncanma-stripe** [comment, marked as answer]
Hey Igor, I have some comments about what we do at Stripe and my own opinion of what you should do 😄 

So first, at Stripe, internal docs are a separate system, stored differently, managed differently (still Markdoc, but with a different set of components and a different workflow for updating them). 

API reference content at Stripe is generated from the source code comments/structure, but has the ability to be overridden. Right now that generated content is checked in, but I would rather it wasn't, since it is generated. In my opinion, source code repos should contain only the original material, not any build output. 

All of the documentation is in a repo with the actual source code, but this is where Stripe's current setup and my opinion differ. I would rather have the markdoc files in a distinct repo, or set of repos. There is the benefit that developers may find it easy to edit the docs if they are in a repo they already use, but switching between multiple repos is pretty normal behavior for devs at many companies, so I don't believe that is really a big issue. Having the markdoc in a content-only repo (or repos) allows you to keep it very small, manage permissions separately, have a documentation specific set of CI, validation, build rules... and it makes the workflow quicker and easier for authoring. At Microsoft, for example, each product's docs are in a distinct repo, so there are 100s of them. 

Oh, and yes, developers update the docs... as do PMs, EMs, and of course the technical writers.

**Comment by rpaul-stripe** [comment]
Our product documentation is in the monorepo, but the internal documentation is split: a small portion of internal docs that engineers want to maintain in git alongside the code are stored in the monorepo while the rest are stored in a custom database-backed Markdoc CMS that has a browser-based frontend. We ended up with this approach because of various internal constraints that are specific to Stripe (we have to share a lot of underlying development infrastructure that is maintained by teams that prioritize engineering use cases over documentation use cases), so what we do today isn't necessarily how we would do things if we had the choice.

The benefits we get from colocating documentation with code are that it makes it easier for engineers to contribute to the content and ship large documentation changes into production alongside the equivalent functional changes to the code. It also makes it easier for us to incorporate Stripe functionality into the content and keep the custom tag implementations tied directly to corresponding changes to the underlying React components without having to worry about versioning. The main downside is that our documentation publishing toolchain is tightly coupled with the same system that Stripe uses to review and deploy code, which means that there's a lot of overhead and unnecessary friction for content changes. This creates a lot of pain points around publishing time and how much work is needed to write, preview, and deploy trivial documentation-only edits.

**Reply by kryltsov** [reply]
Thanks @rpaul-stripe and @duncanma-stripe - very helpful!

**Reply by kryltsov** [reply]
@duncanma-stripe 
If you start from scratch in a team with about 100 repos would it be too risky to have a single repo for all docs internal and external using Markdoc tags to generate internal and external documentation from the same source?

By risky I mean making some block external exposing some stuff by an accident.

On the other side, I see a lot of the same stuff that will need to be exposed both external and internal that will require duplication and sync. Am I overcomplicating this - should they just be two separate repos ...?
